crud/request.go

```diff
@@ -0,0 +1,1129 @@
+
+// Opts sets the options for the UpsertManyRequest request.
+// Note: default value is nil.
+func (req *UpsertManyRequest) Opts(opts interface{}) *UpsertManyRequest {
+	req.opts = opts
+	return req
+}
+
+// Body fills an encoder with the call request body.
+func (req *UpsertManyRequest) Body(res tarantool.SchemaResolver, enc *encoder) error {
+	req.impl.Args([]interface{}{req.space, req.tuplesOperationData, req.opts})
+	return req.impl.Body(res, enc)
+}
+
+// Context sets a passed context to CRUD request.
+func (req *UpsertManyRequest) Context(ctx context.Context) *UpsertManyRequest {
+	req.impl = req.impl.Context(ctx)
+
+	return req
+}
+
+// UpsertObjectManyRequest helps you to create request
+// object to call `crud.upsert_object_many` for execution
+// by a Connection.
+type UpsertObjectManyRequest struct {
+	spaceRequest
+	objectsOperationData interface{}
+	opts                 interface{}
+}
+
+// NewUpsertObjectManyRequest returns a new empty UpsertObjectManyRequest.
+func NewUpsertObjectManyRequest(space interface{}) *UpsertObjectManyRequest {
+	req := new(UpsertObjectManyRequest)
+	req.impl = tarantool.NewCallRequest("crud.upsert_object_many")
+	req.setSpace(space)
+	req.objectsOperationData = []interface{}{}
+	req.opts = map[string]interface{}{}
+	return req
+}
+
+// ObjectOperationData sets objects and operations
+// for the UpsertObjectManyRequest request.
+// Note: default value is nil.
+func (req *UpsertObjectManyRequest) ObjectsOperationData(
+	objectsOperationData interface{}) *UpsertObjectManyRequest {
+	req.objectsOperationData = objectsOperationData
+	return req
+}
+
+// Opts sets the options for the UpsertObjectManyRequest request.
+// Note: default value is nil.
+func (req *UpsertObjectManyRequest) Opts(opts interface{}) *UpsertObjectManyRequest {
+	req.opts = opts
+	return req
+}
+
+// Body fills an encoder with the call request body.
+func (req *UpsertObjectManyRequest) Body(res tarantool.SchemaResolver, enc *encoder) error {
+	req.impl.Args([]interface{}{req.space, req.objectsOperationData, req.opts})
+	return req.impl.Body(res, enc)
+}
+
+// Context sets a passed context to CRUD request.
+func (req *UpsertObjectManyRequest) Context(ctx context.Context) *UpsertObjectManyRequest {
+	req.impl = req.impl.Context(ctx)
+
+	return req
+}
+
+// SelectRequest helps you to create request
+// object to call `crud.select` for execution
+// by a Connection.
+type SelectRequest struct {
+	spaceRequest
+	conditions interface{}
+	opts       interface{}
+}
+
+// NewSelectRequest returns a new empty SelectRequest.
+func NewSelectRequest(space interface{}) *SelectRequest {
+	req := new(SelectRequest)
+	req.impl = tarantool.NewCallRequest("crud.select")
+	req.setSpace(space)
+	req.conditions = []interface{}{}
+	req.opts = map[string]interface{}{}
+	return req
+}
+
+// Conditions sets the conditions for the SelectRequest request.
+// Note: default value is nil.
+func (req *SelectRequest) Conditions(conditions interface{}) *SelectRequest {
```

Comment:
It does not look cool for a statically-typed language [2].

Reply:
PR https://github.com/tarantool/go-tarantool/pull/266